File: scratch_adult.js
const db = require('better-sqlite3')('local_db.sqlite');
const rows = db.prepare("SELECT data FROM responses WHERE pathway = 'adult_circumcised'").all();
const res = rows.map(r => JSON.parse(r.data));
const context = {};
const age = {};
res.forEach(r => {
  const c = r.circ_adult_context;
  context[c] = (context[c] || 0) + 1;
  const a = r.circ_age;
  age[a] = (age[a] || 0) + 1;
});
console.log('CONTEXT', context);
console.log('AGE', age);

File: scratch_adult.py
import sqlite3

conn = sqlite3.connect('local_db.sqlite')
c = conn.cursor()

c.execute("SELECT respondent_id FROM responses WHERE question_id = 'circ_age' AND value_text IN ('Adolescence', 'Adulthood')")
ids = [r[0] for r in c.fetchall()]

ids_str = ','.join(map(str, ids))

c.execute(f"SELECT question_id, value_text FROM responses WHERE respondent_id IN ({ids_str}) AND question_id IN ('circ_adult_context', 'circ_age')")
res = c.fetchall()

context_counts = {}
age_counts = {}

for row in res:
    qid, val = row
    if qid == 'circ_adult_context':
        context_counts[val] = context_counts.get(val, 0) + 1
    elif qid == 'circ_age':
        age_counts[val] = age_counts.get(val, 0) + 1

print('Total Adult/Adolescent N:', len(ids))
print('CONTEXT:', context_counts)
print('AGE:', age_counts)
